libs/cart/src/effects/cart-item.effects.ts

```diff
@@ -65,16 +72,33 @@ export class DaffCartItemEffects<
   add$ = this.actions$.pipe(
     ofType(DaffCartItemActionTypes.CartItemAddAction),
     switchMap((action: DaffCartItemAdd<U>) =>
-      this.driver.add(
-        this.storage.getCartId(),
-        action.input
-      ).pipe(
-        map((resp: V) => new DaffCartItemAddSuccess(resp)),
+      combineLatest([
+				this.store.pipe(select(
+					getDaffCartItemEntitiesSelectors<V, X, T>().selectAllCartItems)
+				),
+				this.driver.add(this.storage.getCartId(), action.input)
+			]
+			).pipe(
+        map(([cartItems, resp]: [T[], V]) => {
+					//find the cart item that is new or that has a different qty and change the state to "new"
+					//I think we also need to attach the cart item id to the action, so the "new" state can be 
+					//changed back to "default" in a "reset" effect below.
+					return new DaffCartItemAddSuccess(resp, cartItemIdThatHasChanged)
+				}),
         catchError(error => of(new DaffCartItemAddFailure('Failed to add cart item')))
       )
     )
+	)
+
+	@Effect()
+  resetCartItemStateAfterAdd$ = this.actions$.pipe(
+		ofType(DaffCartItemActionTypes.CartItemAddSuccessAction),
+		debounceTime(this.cartItemStateDebounceTime),
```

Comment:
Is it sufficient to have a degrading new state for cart items? Don't we need a static reference to the most recently added cart item?

Reply:
I think it's pretty clean to just have a single source of the cart item state. That way the app dev can have really simple access to any given state of the cart item. And I think if we made the state non-degrading, the app dev would need to find some way to reset the state, because I think the use case is usually for a notification of some sort that disappears after some time. Plus, since we now attach the cart item id to the "DaffCartItemAddSuccess" action, the app dev could always add their own static reference to the most recently added cart item without too much trouble. Is there a use case that I'm not thinking of where we would need to have the static reference? I think that for now, we try to keep the whole thing as simple as possible for the app dev until such a use case comes up and we could add the static reference then as a separate part of state. For the use cases that exist now though, I think the degrading state makes sense and will make life for an app dev really simple.

Reply:
Okay sounds good.